### PR TITLE
Keyboard widget: provide variable for shortcut descriptor to actions.

### PR DESCRIPTION
**Pull request by saqimtiaz**
This PR extends the keyboard widget to provide an additional variable `event-key-descriptor` to the actions it invokes.

This makes it possible for a single keyboard widget to handle multiple keyboard shortcuts configured with key descriptors to be handled within a single block of actions.

`KeyboardManager.prototype.checkKeyDescriptor` has been refactored to use a new method `getMatchingKeyDescriptor`, which can also be used independently to get the `keyInfo` matching a given event. This change is backwards compatible.

Docs have been updated with the new variable.

### CODE REVIEW

File: editions/tw5.com/tiddlers/widgets/KeyboardWidget.tid [hunk 1]
```diff
@@ -30,6 +30,7 @@ The content of the `<$keyboard>` widget is rendered normally. The keyboard short
 |!Variables |!Description |
 |`event-key` |The <<.var event-key>> variable contains the character, if possible. eg: `1`. You can experiment with some settings at: [[Key Codes (Example)]] |
 |`event-code` |The <<.var event-code>> variable contains a character description. eg:  `Digit1` instead of `1`. Or `Space` instead of an empty string ` `, which is hard to see|
+|`event-key-descriptor` |The <<.var event-key-descriptor>> variable is available if the keyboard event captured was configured using a [[keyboard shortcut descriptor|Keyboard Shortcut Descriptor]] of the form `((my-shortcut))` which references a configuration tiddler. |
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/widgets/KeyboardWidget.tid:33]: Do we need a `<<.from-version>>` flag here?

**Review by Jermolene — CHANGES_REQUESTED** [review]
Thanks @saqimtiaz one minor comment

File: editions/tw5.com/tiddlers/widgets/KeyboardWidget.tid (hunk 1 above)
Comment (saqimtiaz) [reply editions/tw5.com/tiddlers/widgets/KeyboardWidget.tid:33]: @Jermolene on the line above the table there is a `from-version` flag covering the entire table as all of these variables have been introduced for 5.2.0

**Comment by saqimtiaz** [comment]
This is what the table with the line in question looks like when rendered:
image

**Comment by Jermolene** [comment]
Apologies and thanks @saqimtiaz

**Comment by saqimtiaz** [comment]
No worries @Jermolene and thank you.

**Pull request merged**